### ISSUE TITLE
Auto travel is unreliable in forests while riding a horse

### ISSUE THREAD

**Issue by ymber** [CleverRaven/Cataclysm-DDA#36313]
# Describe the bug
Auto travel tries to go through things that would be passable on foot but not mounted regardless of whether the player is mounted.

# Steps To Reproduce
1. Use auto travel through a stretch of forest while riding a horse

# Expected behavior
Auto travel is aware of what is passable in the player's current mode of travel.

# Versions and configuration
- OS: Linux
    - OS Version: Kernel 5.4.5-arch1-1
- Game Version: 0.D-10565-g1e8f27e [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fictional Guns [no_rivtech_guns],
    No Flaming Weapons [no_flaming_weapons]
]

**Comment by ipcyborg** [comment]
Is that correct that horse can't walk over a mound of dirt with the `MOUNTABLE` flag? The flag description says: `Usable as a point to fire a mountable weapon from`.

**Comment by ymber** [comment]
That is correct. There's a lot of stuff they can't walk over when you're riding them.
